### PR TITLE
Add checks of spec arguments when issuing VCs

### PR DESCRIPTION
**Pull request by przydatek**
Add checking of argument values from credential spec when issuing credentials.

### CODE REVIEW

File: issuer/src/main.rs [hunk 1]
```diff
@@ -1003,11 +1034,36 @@ fn verify_principal_owns_credential(
...
             .map_err(IssueCredentialError::UnauthorizedSubject)?;
     }
     if let Some(group_record) = groups.get(&(group_name.clone(), owner).into()) {
         if let Some(member_record) = group_record.members.get(&user) {
+            let stored_spec: CredentialSpec = OrdCredentialSpec {
+                credential_type: credential_spec.credential_type.clone(),
+                arguments: member_record.vc_arguments.clone(),
+            }
+            .into();
+            if credential_spec.credential_type == "VerifiedAge" {
```
Comment (lmuntaner) [inline issuer/src/main.rs:1047]: Could we use a variable for `"VerifiedAge"` here and in the GROUP_TYPE?

File: issuer/tests/issue_credentials.rs [hunk 2]
```diff
@@ -400,16 +402,120 @@ fn should_prepare_verfied_member_credential_for_authorized_principal() {
     }
 }
 
+#[test]
+fn should_prepare_verified_age_vc_for_weaker_claim() {
```
Comment (lmuntaner) [inline issuer/tests/issue_credentials.rs:406]: What does "weaker_claim" mean?

File: issuer/tests/issue_credentials.rs [hunk 3]
```diff
@@ -400,16 +402,120 @@ fn should_prepare_verfied_member_credential_for_authorized_principal() {
...
+            credential_spec: add_owner(&requested_spec, owner),
+            signed_id_alias: DUMMY_SIGNED_ID_ALIAS.clone(),
+        },
+    )
+    .expect("API call failed");
+    assert_matches!(response, Ok(_));
+}
+
+#[test]
+fn should_fail_prepare_verified_age_vc_for_stronger_claim() {
```
Comment (lmuntaner) [inline issuer/tests/issue_credentials.rs:436]: What does "stronger claim" mean?

**Review by lmuntaner — APPROVED** [review]
LGTM! Thanks!

File: issuer/src/main.rs (hunk 1 above)
Comment (przydatek) [reply issuer/src/main.rs]: You mean a constant, right?  Done.

File: issuer/tests/issue_credentials.rs (hunk 2 above)
Comment (przydatek) [reply issuer/tests/issue_credentials.rs:408]: It is weaker as the lower bound requested for the VC implied by the actual age (if a user is 21, his/her age is at least 18).  I added a comment.

File: issuer/tests/issue_credentials.rs (hunk 3 above)
Comment (przydatek) [reply issuer/tests/issue_credentials.rs:440]: I added also a comment for this test.

**Review by przydatek — COMMENTED** [review]
Thanks!  Will merge when CI is green.

**Pull request merged**